lib/RefBackend/LowerToRefbackrtABI.cpp

```diff
@@ -21,6 +21,12 @@
 using namespace mlir;
 using namespace mlir::NPCOMP;
 
+// Since input/output shapes are not hyper-rectangular we specify 
+// a maximum rank for each input shape such that shapes are padded
+// out to kMaxRank at the ABI boundary. That way we can represent
+// shapes using a traditional DenseElementsAttr.
+static constexpr int kMaxRank = 6;
```

Comment:
Please add comments linking both definitions of kMaxRank.

Reply:
Comments have been added to link both definitions of kMaxRank.